apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/FirestoreMultipleDbTest.cs

```diff
@@ -0,0 +1,66 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Firestore.IntegrationTests
+{
+
+    public class FirestoreMultipleDbTest
+    {
+        private readonly MulitpleFirestoreDbFixture _fixture;
+        public FirestoreMultipleDbTest()
+        {
+            _fixture = new MulitpleFirestoreDbFixture();
+        }
+        
+        [Fact]
+        public async Task MultipleDbReadWriteTest()
+        {
+            string dbIdPrefix = IdGenerator.FromGuid(prefix: "db", separator: "-");
+            string correctDbId = dbIdPrefix +"correct";
+            string incorrectDbId = dbIdPrefix + "incorrect";
+            string databaseLocationId = "us-east1";
+            string databaseType = "FIRESTORE_NATIVE";
+            // Create two databases one for reading and writing, One for reading to validate that correct database should be referred.
+            await _fixture.CreateDatabaseAsync(correctDbId, databaseLocationId, databaseType);
+            await _fixture.CreateDatabaseAsync(incorrectDbId, databaseLocationId, databaseType);
+            // Write data to the first database; refered as correct database.
+            FirestoreDb correctDbWriter = new FirestoreDbBuilder { DatabaseId = correctDbId, ProjectId = _fixture.ProjectId }.Build();
+            await correctDbWriter.Collection("cities").Document("SF").SetAsync(new Dictionary<string, object>(){
+                { "Name", "San Francisco" },
+                { "State", "CA" },
+                { "Country", "USA" },
+                { "Capital", false },
+                { "Population", 860000 }
+            });
+            // Read data from database that was referenced while writing.
+            FirestoreDb correctDbReader = new FirestoreDbBuilder { DatabaseId = correctDbId, ProjectId = _fixture.ProjectId }.Build();
+            DocumentReference readerDocRef = correctDbReader.Collection("cities").Document("SF");
+            DocumentSnapshot CorrectDbSnapshot = await readerDocRef.GetSnapshotAsync();
+            Dictionary<string, object> items = CorrectDbSnapshot.ToDictionary();
+            // Try to read data from random incorrect database. To validate that the snapshot and data should not exists.
+            FirestoreDb incorrectDbReader = new FirestoreDbBuilder { DatabaseId = incorrectDbId, ProjectId = _fixture.ProjectId }.Build();
+            DocumentReference incorrectDbDocRef = incorrectDbReader.Collection("cities").Document("SF");
+            DocumentSnapshot incorrectDbSnapshot = await incorrectDbDocRef.GetSnapshotAsync();
+
+            Assert.Equal(true, CorrectDbSnapshot.Exists);

```

Comment:
I'd also suggest moving the assertions to just after you've done the conversion.
(In some ways it would be better for this to be two tests, but I can see why it's only one.)